### PR TITLE
[UPnP] Player: Make transport state thread safe

### PR DESCRIPTION
**Pull request by enen92**
## Description
Currently the access to the transport state done by the UPnP Player is not thread safe. `DoAudioWork()` is called by CApp `Process()` to "do stuff" in a timely manner (yeap this is something that should be removed/improved but definitely not for Omega). One of the things it does as part of each iteration is checking if the transport state is stopped to flag end of playback. The state by default is "STOPPED" (even if not playing) and is changed internally by the library using callbacks.
If for some reason the target system takes a bit to download the file and flag the callback it can happen that `DoAudioWork` is called before the callback...killing the playback before it has even started...
This is not deterministic at all, sometimes I could send the files via UPnP other times I couldn't. Wasted quite a bit of time in git bisect, hence added a log line to avoid the pain to the next one :)

I'd like to have this in beta 2 if possible.

This PR makes sure the state is accessed consistently in the player (on playback start and other parts of the class the state is already being accessed the same way) and that the lock is always set on the delegate before getting the state.

## Motivation and context
I have one device (android phone) which I couldn't use to send files via UPnP to remote players.

## How has this been tested?
Runtime tested on Linux and android using Kodi as the upnp player to both kodi (or my tv) as the target.

## What is the effect on users?
Playback via UPnP should be deterministic, shouldn't be stopped before it has even started :)

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/network/upnp/UPnPPlayer.cpp [hunk 1]
```diff
@@ -525,11 +524,14 @@ void CUPnPPlayer::DoAudioWork()
...
-    if(data == "STOPPED")
     {
-      m_started = false;
-      m_callback.OnPlayBackEnded();
+      std::unique_lock<CCriticalSection> lock(m_delegate->m_section);
+      if (m_delegate->m_trainfo.cur_transport_state == "STOPPED")
+      {
+        m_logger->info("Transport state flagged as STOPPED. Triggering OnPlayBackEnded.");
+        m_started = false;
+        m_callback.OnPlayBackEnded();
```
Comment (garbear) [inline xbmc/network/upnp/UPnPPlayer.cpp:533]: Is this a blocking callback?

Comment (enen92) [reply xbmc/network/upnp/UPnPPlayer.cpp:533]: Yeah it is trapped by python and others.
I see your point, will move it out of the scoped lock 👍

Thanks

Comment (enen92) [reply xbmc/network/upnp/UPnPPlayer.cpp]: Improved on the last revision. Runtime tested again.

**Comment by enen92** [comment]
Rebased, hopefully still in time for beta 2

**Pull request merged**